jujugui/static/gui/src/app/jujulib/test-baker-factory.js

```diff
@@ -0,0 +1,46 @@
+/* Copyright (C) 2016 Canonical Ltd. */
+
+'use strict';
+
+chai.config.includeStack = true;
+chai.config.truncateThreshold = 0;
+
+describe('jujulib bakery factory', function() {
+
+  var fakeBakery = function(cfg) {
+    this.serviceName = cfg.serviceName;
+  };
+
+  fakeBakery.prototype.clearCookie = sinon.stub();
+
+  var makeFactory = function() {
+    let cfg = { Bakery: fakeBakery };
+    return new window.jujulib.bakeryFactory(cfg);
+  };
+
+  it('exists', function() {
+    let factory = makeFactory();
+    assert.strictEqual(factory instanceof window.jujulib.bakeryFactory, true);
+  });
+
+  it('can create a bakery', function() {
+    let factory = makeFactory();
+    factory.create({serviceName: 'foo'});
+    assert.isTrue(factory.bakeries.has('foo'));
+  });
+
+  it('can get a bakery', function() {
+    let factory = makeFactory();
+    let cfg = {serviceName: 'foo'};
+    factory.create(cfg);
+    let bakery = factory.get(cfg);
+    assert.equal(bakery.serviceName, 'foo');
+  });
+
+  it('can clear the cookies for a bakery', function() {
+    let factory = makeFactory();
+    factory.create({serviceName: 'foo'});
+    factory.clearAllCookies();
+    assert.isTrue(factory.bakeries.get('foo').clearCookie.called);
+  });
+});
```

Comment:
What happens if we use the same name to create a new bakery? I guess the second one overrides the first one. Let's document this behavior with a test?